### ISSUE TITLE
Babel requires preset

### ISSUE THREAD

**Issue by amsross** [choojs/bankai#341]
`bankai start` returns this error:
```
scripts:browserify.bundle Couldn't find preset "es2015" relative to directory "/app/client/node_modules/hawk" while parsing file: /app/client/node_modules/hawk/dist/browser.js
Error: Couldn't find preset "es2015" relative to directory "/app/client/node_modules/hawk" while parsing file: /app/client/node_modules/hawk/dist/browser.js
    at /app/client/node_modules/babel-core/lib/transformation/file/options/option-manager.js:293:19
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (/app/client/node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
    at OptionManager.mergePresets (/app/client/node_modules/babel-core/lib/transformation/file/options/option-manager.js:264:10)
    at OptionManager.mergeOptions (/app/client/node_modules/babel-core/lib/transformation/file/options/option-manager.js:249:14)
    at OptionManager.init (/app/client/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/app/client/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/app/client/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (/app/client/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at Babelify._flush (/app/client/node_modules/babelify/index.js:26:24)
```

By adding `babel-presets-es2015` (`npm install --save-dev babel-presets-es2015`), I was able to successfully start the dev server.

**Comment by goto-bus-stop** [comment]
Hmm, it should not try to transform code from node_modules with custom presets, only with babel-preset-env. I think we need to add `babelrc: false` to the babelify config in `lib/graph-script.js`, so that .babelrc files in node_modules are not picked up.

**Comment by amsross** [comment]
@goto-bus-stop that addresses the issue. I'll take a look at why that works and open a PR for it.

Thanks!

**Comment by toddself** [comment]
There is something more going on here and we might have to think about this some more.

Example: [diff](https://github.com/kpdecker/jsdiff), when you install from npm has a re-built version to use in the browser.

This is build, in part, by using babel.  So, to configure babel, there is a `babel` section in the package.json, but the presets are stored as devDepenendencies (correctly since they're not needed for the installed version).

However, when babelify runs globally, it is trying to run the babel transforms specified by the babel section in the package.json.  

This fails since the presets aren't installed, and shouldn't be happening anyway since we're already consuming a pre-built file.

This is a bit of a conundrum.

We need to let people provide custom babel configuration through babelrc/package.json, but we shouldn't be running babel on packages that have already had it run and are giving a compiled version of their code out.

**Comment by toddself** [comment]
diff shouldn't have to remove their `babel` section from their package.json (since they are relying on this to make their build), but they shouldn't have to make everyone eat the disk cost of what is, essentially, a development dependency (the presets)

**Comment by amsross** [comment]
The babelify docs warn that using `global: true` will include the `node_modules` directory. They suggest using the `ignore` flag to avoid that.

**Comment by amsross** [comment]
I guess my next question is "why was `global: true` included in https://github.com/choojs/bankai/commit/c45c23ffb507b0d3e2b19b535f00bff7723aea8e?" According to [the `babelify` docs](https://github.com/babel/babelify#why-arent-files-in-node_modules-being-transformed), not including `node_modules` is the default behavior, which is overridden here with `global: true`. It seems to me that once that option is provided, `babelify`'s expectation is that you will provide fairly fine-grained options around not globbing in undesired scripts. Doing so seems impossible in a tool/library vs an app.

**Comment by toddself** [comment]
This was part of the discussion around adding babelify to bankai @amsross; about how to do it safely.

**Comment by amsross** [comment]
I agree with what you're [saying about not needing/wanting to rebuild jsdiff](https://github.com/choojs/bankai/issues/341#issuecomment-346706021) @toddself. It seems to me, though, that whatever is the requirement of the lowest common denominator of all of the dependencies becomes necessary for all dependencies.

For example, consider that you have 100 dependencies that have all been run through babel, etc, but you have 1 dependency that is not (but has a babel config provided in some way). The difference between the two groups would be opaque to bankai, so you would need to rebuild all 101 dependencies. Is that correct?

Again, I don't have any deep knowledge of babel, browserify, or babelify, so maybe I'm limiting my thinking to only considering the `global` a `babelrc` babelify options.

**Comment by toddself** [comment]
@amsross the problem is that you don't know what the dependencies are.  Diff is a dep of a dep here -- we don't know what we're missing until we try to build in bankai for the first time; there is zero way to discover this until that point.

The issue is running babel globally across your modules that also use babel is going to break unless those modules list their babel packages as dependencies instead of devDependencies, or distribute their stuff in source-only mode and make you build them locally.

**Comment by amsross** [comment]
Appears to be resolved by #352
